[CS2103T-T09-1] Plan²Travel

OneArmyj opened the pull request:
@ernestyyh 
@kxxmxxx
@oscarsu97
@1nefootstep

joshuaseetss commented on docs/UserGuide.adoc:
```diff
@@ -54,108 +53,226 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+
+=== Unscheduling activities from a day: `unschedule activity`
+
+Unschedule activities from a day +
```
Unscheduled can be slightly misleading. Perhaps use deleteSchedule?

joshuaseetss commented on docs/UserGuide.adoc:
```diff
@@ -14,17 +14,17 @@ ifdef::env-github[]
+.  Copy the file to the folder you want to use as the home folder for your Plan^2^Travel.
 .  Double-click the file to start the app. The GUI should appear in a few seconds.
 +
 image::Ui.png[width="790"]
```
Will there be an accommodations feature as shown in the picture?

joshuaseetss commented on docs/UserGuide.adoc:
```diff
@@ -54,108 +53,226 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+
+* `edit activity 1 n/Dinner d/0.5 l/Hotel`
+
+Edits the name, duration and location of the 1st activity to be Dinner, 0.5 and Hotel respectively.
```
Perhaps there could be better indentation to this line, to make it clear that it is the description of the given example.

joshuaseetss reviewed (commented):
Very Neat and organised User Guide!

anqichen9856 commented on docs/DeveloperGuide.adoc:
```diff
@@ -242,6 +247,104 @@ _{Explain here how the data encryption feature will be implemented}_
+==== Design considerations
+Aspect: How version comparison is executed:
+
+* Alternative 1(current choice): user overload each time slot.
```
Would be good if you could add justification of why you finally chose this alternative over the other one. Since both have pros and cons, how did you weigh them?

anqichen9856 commented on docs/diagrams/OptimiseSequenceDiagram.puml:
```diff
@@ -0,0 +1,82 @@
+OptimiseCommandParser -> OptimiseBudgetCommand
+activate OptimiseBudgetCommand
+
+OptimiseBudgetCommand --> OptimiseCommandParser
```
Should have a returned value like "o:OptimiseBudgetCommand"

anqichen9856 commented on docs/diagrams/OptimiseSequenceDiagram.puml:
```diff
@@ -0,0 +1,82 @@
+OptimiseCommandParser -> OptimiseCommandParser
+deactivate OptimiseCommandParser
+
+OptimiseCommandParser --> PlannerParser
```
Should have a returned value like "o:OptimiseBudgetCommand"

anqichen9856 commented on docs/diagrams/OptimiseSequenceDiagram.puml:
```diff
@@ -0,0 +1,82 @@
+OptimiseCommandParser -[hidden]-> PlannerParser
+destroy OptimiseCommandParser
+
+PlannerParser --> LogicManager
```
Should have a returned value like "o:OptimiseBudgetCommand"

anqichen9856 commented on docs/diagrams/OptimiseSequenceDiagram.puml:
```diff
@@ -0,0 +1,82 @@
+participant ":PlannerParser" as PlannerParser LOGIC_COLOR
+participant ":OptimiseCommandParser" as OptimiseCommandParser LOGIC_COLOR
+participant ":OptimiseBudgetCommand" as OptimiseBudgetCommand LOGIC_COLOR
+participant ":CommandResult" as CommandResult LOGIC_COLOR
```
CommandResult might not be necessary for this sequence diagram; it's extra detail that does not really affect the understanding of how optimisation works

anqichen9856 commented on docs/diagrams/ScheduleActivityCommandCreation.puml:
```diff
@@ -0,0 +1,60 @@
+PlannerParser --> LogicManager : a
+deactivate PlannerParser
+
+LogicManager -> ScheduleActivityCommand : execute()
```
Would be better to use a reference frame to the next diagram since they're inherently linked

anqichen9856 reviewed (commented):
Good effort! Needs more update on architecture level diagrams.

1nefootstep committed:
Attribute AutoCompleteTextField to original writer

1nefootstep committed:
Add list day INDEX command

1nefootstep committed:
Add list day to suggestions

oscarsu97 committed:
Fix bug

Not able to schedule same activity in same timeslot now

oscarsu97 committed:
Add limit to priority

Range from 1 to 7

oscarsu97 committed:
Fix failing test

KxxMxxx committed:
Merge pull request #234 from oscarsu97/branch-duplicate-schedule-activity

Fix bug of allowing same activity to schedule in same timeslot

KxxMxxx committed:
Merge pull request #236 from oscarsu97/branch-priority

Add limit to priority

Commit:
fix bugs related to contact

ernestyyh committed:
Merge branch 'master' into enhance-storage

Commit:
change test for weaker isSameContact

OneArmyj committed:
Update DG Undo/Redo(WIP)

OneArmyj committed:
Merge pull request #232 from 1nefootstep/fix-schedule-unschedule-autocomplete

Add schedule/unschedule to autocomplete

OneArmyj committed:
Merge pull request #233 from 1nefootstep/add-list-day-command

Add list day command

OneArmyj committed:
Merge pull request #231 from KxxMxxx/fix-bugs

Fix bugs

KxxMxxx committed:
Merge pull request #230 from OneArmyj/update-undo

Fix Undo/Redo Bug

OneArmyj committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T09-1/main

OneArmyj committed:
Edit UML

OneArmyj committed:
Edit DG for undo redo

ernestyyh committed:
Merge branch 'master' into enhance-storage

OneArmyj committed:
Update UG and PPP

OneArmyj committed:
Fix checkstyle

ernestyyh committed:
Merge pull request #238 from OneArmyj/master

Update UG/DG (WIP)

OneArmyj committed:
Edit ppp

OneArmyj committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T09-1/main

OneArmyj committed:
Edit ppp

ernestyyh committed:
Merge branch 'master' into enhance-storage

OneArmyj committed:
Fix Undo Redo bug

OneArmyj committed:
fix checkstyle

OneArmyj committed:
fix test

Commit:
update UG/DG

Commit:
update UG/DG

1nefootstep committed:
Fix delete day bug (#242)

OneArmyj committed:
Fix implementation of undo/redo for all Add, Delete and Edit commands

OneArmyj committed:
Fix implementation of undo/redo for all Add, Delete and Edit commands

oscarsu97 committed:
Limit number of overlapping activities to 5 (#241)

* Limit number of overlap to 5

* Fix checkStyle

* Fix code quality

OneArmyj committed:
Fix undo redo bugs

OneArmyj committed:
Merge branch 'fix-undo-redo' of https://github.com/OneArmyj/main into fix-undo-redo

OneArmyj committed:
fix checkstyle

OneArmyj committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T09-1/main into fix-undo-redo

OneArmyj committed:
Fix delete day bug, fix schedule undo bug

OneArmyj committed:
Add command in UndoableCommand for abstraction

Commit:
Merge pull request #243 from OneArmyj/fix-undo-redo

Fix undo redo bugs

OneArmyj committed:
Fix clear command and optimise command

KxxMxxx committed:
Merge pull request #244 from OneArmyj/fix-undo-redo

Fix clear command and optimise command

ernestyyh committed:
Merge branch 'master' into enhance-storage

OneArmyj committed:
Edit undo redo UML diagrams

Commit:
update hashmap to reduce duplicate contacts

OneArmyj committed:
Edit UserGuide, DevGuide and ppp

OneArmyj committed:
Delete DeveloperGuide.pdf

OneArmyj committed:
Merge pull request #245 from KxxMxxx/fix-bugs

update hashmap to reduce duplicate contacts

ernestyyh committed:
Merge branch 'master' into enhance-storage

oscarsu97 committed:
Update UG

oscarsu97 committed:
Update UG for autoschedule command

1nefootstep committed:
Add ResultInformationTest

1nefootstep committed:
Update CommandResultTest

1nefootstep committed:
Fix DeleteContactCommandTest

Commit:
update UG/DG/PPP

Commit:
update UG/DG/PPP

KxxMxxx committed:
Merge pull request #246 from OneArmyj/update-ppp

Update ppp

Commit:
update hashmap

now, if nothing shares the contact, when editing the phone of
activity/accommodation, the contact would be edited as well

ernestyyh committed:
Merge branch 'master' into enhance-storage

Commit:
fix checkstyle

OneArmyj committed:
Merge pull request #248 from KxxMxxx/ppp

Update UG/PPP

Commit:
update edit activity/accommodation command

OneArmyj committed:
Merge pull request #249 from KxxMxxx/fix-bugs

update hashmap

ernestyyh committed:
Merge branch 'master' into enhance-storage

oscarsu97 committed:
Update schedule activity feature in dg

Commit:
fix duplicate contact bug

OneArmyj committed:
Merge pull request #250 from KxxMxxx/fix-bugs

fix duplicate contact bug

OneArmyj committed:
Edit ModelManager

OneArmyj committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T09-1/main

1nefootstep committed:
Update PPP

1nefootstep committed:
Update PPP

1nefootstep committed:
Format PPP

1nefootstep committed:
Update UG

1nefootstep committed:
Fix images

OneArmyj committed:
Change EditActivity and EditAccommodation Commands

oscarsu97 committed:
Update schedule feature in dg

OneArmyj committed:
fix checkstyle

ernestyyh committed:
Merge branch 'master' into enhance-storage

KxxMxxx committed:
Merge pull request #251 from OneArmyj/master

Edit commands

OneArmyj committed:
fix bugs for duplicate contacts

OneArmyj committed:
fix checkstyle

1nefootstep committed:
Update modelclassdiagram

1nefootstep committed:
Add InfoListPanel

OneArmyj committed:
Fix mapping for undo redo execution

1nefootstep committed:
Add HelpListPanel

KxxMxxx committed:
Merge pull request #252 from OneArmyj/master

fix mappings of Contact to Model

Commit:
fix delete contact bug

1nefootstep committed:
Add AgendaManager

1nefootstep committed:
Fix checkstyle

OneArmyj committed:
Add authorship for OneArmyj

1nefootstep committed:
Update DG

Commit:
fix contact bugs

Commit:
fix bugs

OneArmyj committed:
Merge pull request #253 from KxxMxxx/fix-bugs

fix delete contact bug

OneArmyj committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T09-1/main

1nefootstep committed:
Change from InitCommand to SetCommand

Commit:
Merge branch 'master' into ppp

Commit:
Merge branch 'master' into ppp

1nefootstep committed:
Update UG for SetCommand

1nefootstep committed:
Fix checkstyle

Commit:
add authorship

1nefootstep committed:
Add link to user guide

1nefootstep committed:
Change text of link to guide

1nefootstep committed:
Fix checkstyle

OneArmyj committed:
fix contact mapping for undo delete contact

1nefootstep committed:
Add authorship

1nefootstep committed:
Add more authorship

1nefootstep committed:
Change authorship formatting

1nefootstep committed:
Fix formatting

ernestyyh committed:
Merge pull request #256 from 1nefootstep/change-init-to-set

Refactor InitCommand to SetCommand

1nefootstep committed:
Add UnscheduleCommandParser authorship

ernestyyh committed:
Merge branch 'master' into enhance-storage

OneArmyj committed:
Merge pull request #254 from 1nefootstep/ug-dg

Improve code quality of CentralDisplay and update PPP

ernestyyh committed:
Merge branch 'master' into enhance-storage

oscarsu97 committed:
Add autoschedule activity diagram

1nefootstep committed:
Add some logging

1nefootstep committed:
Update UG

1nefootstep committed:
Change from unsigned to signed

ernestyyh committed:
Update Set command

OneArmyj committed:
Fix mapping of adding/deleting model with contact, link is restored

KxxMxxx committed:
Merge pull request #255 from OneArmyj/master

Add authorship & fix linking

OneArmyj committed:
Merge pull request #257 from KxxMxxx/ppp

indicate authorship

OneArmyj committed:
Merge pull request #260 from 1nefootstep/fix-unsigned-int-bug

Fix error in invalid message

OneArmyj committed:
Merge pull request #258 from 1nefootstep/authorship

Add authorship

1nefootstep committed:
Add check to see if index is out of bounds

1nefootstep committed:
Fix checkstyle

oscarsu97 committed:
Update autoschedule command in dg

1nefootstep committed:
Fix bug where itinerary does not appear to clear

1nefootstep committed:
Fix checkstyle

ernestyyh committed:
Implement NewCommand

OneArmyj committed:
Merge pull request #261 from 1nefootstep/fix-list-day-out-of-bound

Fix list day bug

1nefootstep committed:
Fix UnscheduleCommand invalid message

OneArmyj committed:
update ppp and dg

OneArmyj committed:
Merge pull request #262 from 1nefootstep/fix-ui-bugs

Fix UI bug where clear does not remove the agenda from agendaTab

Commit:
add equals method for optimisecommand

Commit:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T09-1/main

OneArmyj committed:
fix tag

OneArmyj committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T09-1/main

ernestyyh committed:
Merge branch 'master' into enhance-storage

OneArmyj committed:
Merge pull request #264 from KxxMxxx/master

Add equals method for OptimiseCommand

1nefootstep committed:
Fix unschedule out of bound bug

1nefootstep committed:
Edit grammar

Commit:
update undoclearcommand to update ui

ernestyyh committed:
Update ClearCommand and NewCommand

oscarsu97 committed:
Update PPP

KxxMxxx committed:
Merge pull request #263 from 1nefootstep/edit-messages

Fix UnscheduleCommand bugs

KxxMxxx committed:
update undoclearcommand to update ui (#265)

oscarsu97 committed:
Fix code quality

oscarsu97 committed:
Add new line at EOF

1nefootstep committed:
Fix delete day bug

KxxMxxx committed:
Merge pull request #266 from oscarsu97/branch-ugdg

update UG & DG

oscarsu97 committed:
Fix bug in autoschedule

oscarsu97 committed:
Add authorship

Commit:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T09-1/main

Commit:
update/add test and update modelmanager

Commit:
fix naming of optimise test method

Commit:
update activity test

Commit:
update optimise test method names

ernestyyh committed:
Implement LoadCommand

Case of AccessDeniedException has not been resolved

OneArmyj committed:
Fix undo for delete day

OneArmyj committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T09-1/main

OneArmyj committed:
Merge pull request #267 from 1nefootstep/fix-delete-day-bug

Fix delete day bug

OneArmyj committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T09-1/main

OneArmyj committed:
Fix undo bug for delete day (#269)

* update ppp and dg

* fix tag

* Fix undo for delete day

ernestyyh committed:
Enhance storage (#270)

* Implement renamable single folder system

* Update Set command

* Implement NewCommand

* Update ClearCommand and NewCommand

* Implement LoadCommand

Case of AccessDeniedException has not been resolved

ernestyyh committed:
Merge branch 'master' into enhance-storage

oscarsu97 committed:
Update UG

oscarsu97 committed:
Fix image not showing in DG

oscarsu97 committed:
Update ppp

OneArmyj committed:
Merge pull request #268 from KxxMxxx/testing

update/add test and update modelmanager

oscarsu97 committed:
Edit autoschedulecommand message usage

OneArmyj committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T09-1/main

OneArmyj committed:
Delete onearmyj.pdf

oscarsu97 committed:
Shorten edit activity message usage

oscarsu97 committed:
Shorten message usage for contact and acccommodation

oscarsu97 committed:
Remove traces of code from AB3

OneArmyj committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T09-1/main

oscarsu97 committed:
Update UG & DG

oscarsu97 committed:
Make minor touch up on ppp

OneArmyj committed:
Add logging + Javadoc

ernestyyh committed:
Merge pull request #272 from OneArmyj/logging-branch

Logging branch

oscarsu97 committed:
Add newline at EOF

oscarsu97 committed:
Fix spelling error

OneArmyj committed:
update ug

oscarsu97 committed:
Fix image not showing

oscarsu97 committed:
Remove spacing in command format

Commit:
fix delete day bug

1nefootstep committed:
Partially update DG

oscarsu97 committed:
Add limit to scheduling activity in same timeslot

ernestyyh committed:
Update and fix bugs in LoadCommand, NewCommand and SetCommand

ernestyyh committed:
Merge branch 'master' into enhance-storage

1nefootstep committed:
Remove extra shifting of dates

Commit:
Revert "Delete onearmyj.pdf"

This reverts commit 4ea8d9b7a5be60f67c1caf77447499470c3fe104.

1nefootstep committed:
Remove commented out code

oscarsu97 committed:
Remove traces of AB3

KxxMxxx committed:
Merge pull request #275 from 1nefootstep/fix-delete-day-2

Remove extra shifting of dates

Commit:
Merge branch 'master' into ppp

OneArmyj committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T09-1/main

OneArmyj committed:
Merge pull request #274 from ernestyyh/enhance-storage

Update and fix bugs in LoadCommand, NewCommand and SetCommand

OneArmyj committed:
Add UndoCommandTest (WIP)

OneArmyj committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T09-1/main

Commit:
update ppp/ug/dg

OneArmyj committed:
Merge pull request #247 from 1nefootstep/add-test

Add Tests

OneArmyj committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T09-1/main

Commit:
fix accidental changes

Commit:
Merge branch 'master' into ppp

Commit:
refactor resultinformationtest

OneArmyj committed:
Add tests for UndoCommand and RedoCommand

OneArmyj committed:
Merge pull request #276 from KxxMxxx/ppp

update PPP/UG/DG

OneArmyj committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T09-1/main

KxxMxxx committed:
Merge pull request #277 from OneArmyj/master

Add Tests for UndoCommand and RedoCommand

ernestyyh committed:
Implement CopyToCommand

ernestyyh committed:
Merge branch 'master' into enhance-storage

1nefootstep committed:
Update DG

1nefootstep committed:
Shift timetable up

OneArmyj committed:
Merge pull request #278 from ernestyyh/enhance-storage

Implement CopyToCommand

1nefootstep committed:
Update DG

1nefootstep committed:
Update PPP

1nefootstep committed:
Fix include

oscarsu97 committed:
Finalise PPP

KxxMxxx committed:
Merge pull request #271 from oscarsu97/branch-fix-bugs2

fix bugs for autoschedule

1nefootstep committed:
Update PPP

OneArmyj committed:
Add tests for Add/Delete/Clear Event package

OneArmyj committed:
Merge pull request #279 from 1nefootstep/DG-autocomplete

Update DG UG PPP

OneArmyj committed:
Add tests for Add/Delete/Clear Event package (#280)

Commit:
implement test for addcommands

Commit:
add logger to optimise command

OneArmyj committed:
Merge pull request #281 from KxxMxxx/testing

implement test for addcommands

1nefootstep committed:
Update DG (#282)

* Update DG

* Remove product survey

* Remove pdfs

OneArmyj committed:
Merge pull request #283 from KxxMxxx/master

add logger to optimise command

ernestyyh committed:
Update UG

1nefootstep committed:
Add test cases

oscarsu97 committed:
Add manual test

OneArmyj committed:
Fix UG

oscarsu97 committed:
Add logger

OneArmyj committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T09-1/main

OneArmyj committed:
Merge pull request #285 from 1nefootstep/DG-add-test

Add manual test cases to DG

OneArmyj committed:
Merge pull request #287 from OneArmyj/master

Fix UG

OneArmyj committed:
Merge pull request #284 from ernestyyh/document-submission

Update UG

OneArmyj committed:
Merge pull request #286 from oscarsu97/branch-fix-bugs3

add manual test to dg

OneArmyj committed:
fix help

ernestyyh committed:
Update documents and authorship

ernestyyh committed:
Updated size

OneArmyj committed:
Merge pull request #288 from ernestyyh/late-submission

Update documents and authorship

ernestyyh committed:
Update bulleting

ernestyyh committed:
Update PPP

ernestyyh committed:
Merge pull request #289 from ernestyyh/late-submission

Update PPP
